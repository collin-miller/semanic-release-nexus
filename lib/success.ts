export const success = () => {
    // eslint-disable-next-line no-console
    console.log('This is success.');
};
