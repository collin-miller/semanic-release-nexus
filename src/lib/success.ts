import { Context } from 'semantic-release';
import { IPluginConfig } from './plugin';

export const success = async (_pluginConfig: IPluginConfig, context: Context) => {
    const { logger } = context;
    logger.log('This is success.');
};
