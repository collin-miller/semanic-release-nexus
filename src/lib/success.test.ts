import { afterEach, beforeEach, describe, expect, it, jest } from '@jest/globals';
import { success } from './success';

describe('success', () => {
    beforeEach(() => {
        jest.spyOn(console, 'log');
    });
    afterEach(() => {
        jest.restoreAllMocks();
    });

    it('should log the success message', () => {
        success();
        // eslint-disable-next-line no-console
        expect(console.log).toHaveBeenCalledTimes(1);
        // eslint-disable-next-line no-console
        expect(console.log).toHaveBeenCalledWith('This is success.');
    });
});
