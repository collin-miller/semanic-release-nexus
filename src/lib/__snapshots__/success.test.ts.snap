// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`success should log the success message 1`] = `
Array [
  Array [
    "This is success.",
  ],
]
`;
